src/osd/PrimaryLogPG.cc

```diff
@@ -13413,10 +13093,7 @@ int PrimaryLogPG::prep_backfill_object_push(
     h);
   if (r < 0) {
     dout(0) << __func__ << " Error " << r << " on oid " << oid << dendl;
-    primary_failed(oid);
-    primary_error(oid, v);
-    backfills_in_flight.erase(oid);
-    missing_loc.add_missing(oid, v, eversion_t());
+    on_failed_pull({ pg_whoami }, oid, v);
```

Comment:
(PrimaryLogPG.cc:13096) Is the idea here to use `on_failed_pull()`->`force_object_missing()`->`missing_loc.rebuild()` to do an equivalent of 
`missing_loc.add_missing(oid, v, eversion_t());` to update the `needs_recovery_map`?

Reply:
@neha-ojha Yeah, that's the notion.  I'm trying to reduce the number of ways in which the missing set gets messed with as it was proving tricky to thread into PeeringState sensibly.